target/linux/ath79/dts/ar9132_buffalo_wzr-hp-g300nh.dts

```diff
@@ -0,0 +1,259 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9132.dtsi"
+
+/ {
+	compatible = "buffalo,wzr-hp-g300nh", "qca,ar9132";
+	model = "Buffalo WZR-HP-G300NH";
+
+	aliases {
+		led-boot = &led_diag;
+		led-failsafe = &led_security;
+		led-upgrade = &led_diag;
+	};
+
+	clock40mhz: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <40000000>;
+	};
+
+	gpio2: gpio-extender {
+		compatible = "nxp,74hc153-gpio";
+		gpio-controller;
+		#gpio-cells = <2>;
+
+		// GPIOs used by this node
+		gpio-s0 = <&gpio 9 GPIO_ACTIVE_HIGH>;
+		gpio-s1 = <&gpio 11 GPIO_ACTIVE_HIGH>;
+		gpio-1y = <&gpio 12 GPIO_ACTIVE_HIGH>;
+		gpio-2y = <&gpio 14 GPIO_ACTIVE_HIGH>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		aoss {
+			label = "aoss";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+/*
+		router_on {
+			label = "router_on";
+			linux,code = <BTN_5>;
+			gpios = <&gpio2 2 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+
+		qos_off {
+			label = "qos_off";
+			linux,code = <BTN_3>;
```

Comment:
Any reason why you use different number here compared to ar71xx?

Reply:
Ah, you didn't switch the BTN_x, but also the gpios are switched, so effectively qos_off and qos_on is swapped compared to ar71xx. Is this by intention (then please add the info to commit message) or by accident (then, just change label and node name back).

Reply:
Yes, it's by intention as qos_off was pressed when the button was in 'On' state. I think I mentioned it already it the commit message. Should I be more elaborate?

Reply:
Yes, only the gpio-labels are changed as the one labelled 'qos_off' was pressed and high when the button was 'On'. I think I mentioned that in the commit.

Reply:
So, is this the "Movie Engine" comment? Why is it called qos here?

Reply:
It says 'Movie Engine' on the cover. Someone making the ar71xx support translated it to 'qos'

Reply:
So, maybe let's just call it "movie_on" and "movie_off" here?

Reply:
OT: Just consider this a new chance to make things right. ar71xx can be used nicely as a blueprint, but after all to goal is to have proper device support in ath79, where consistency with ar71xx is secondary IMO if we can have it "more correct" now.

Reply:
I dropped the keys for now. I'll rename them in future once the issue with sliding toggles comes to a conclusion.